### DISCUSSION TITLE
Adding a "Configure Table" Button to `htmx_table` Rendering

### DISCUSSION THREAD

**Discussion by Kani999** [netbox-community/netbox#16898] [Q&A]
Hello everyone,

I'm currently using the following code snippet to render an `htmx_table` in my Django project:
`{% htmx_table 'circuits:provideraccount_list' provider_id=object.pk %}`

This works perfectly for rendering the table. However, I would like to add a "Configure Table" button alongside it, as shown in the image below:

![Configure Table Button Example](https://github.com/user-attachments/assets/0f460b4c-694a-46ed-802a-06e0b08d0929)

The purpose of this button is to allow users to configure which columns are visible in the table. 

**Question:**  
How can I implement this functionality? Specifically, I’d like guidance on:  
1. Adding the button next to the table.  
2. Linking it to a modal or form where users can select visible columns dynamically.  

Any suggestions or examples (especially using Django, `htmx`, or `django-tables2`) would be greatly appreciated!

Thank you in advance!

**Comment by llamafilm** [comment]
Did you ever find an answer?
I’m curious what you are using htmx_table for, I’ve never seen that documented. Is this part of an export template?

**Reply by Kani999** [reply]
I'm trying to implement a "Configure Table" button using `htmx_table` in my Netbox plugin. I'm leveraging the `table_controls_htmx.html` template for this, which includes the button and modal for column configuration.

Here's the relevant snippet from my `generic_tab_list.html` template which I'm using at [netbox-attachment plugin](https://github.com/Kani999/netbox-attachments/blob/90d046d55636cdff54531d3207e33fd4c4c476d3/netbox_attachments/templates/netbox_attachments/generic_tab_list.html#L12):

```
{% block content %}
    {% block table_controls %}
        {% include "inc/table_controls_htmx.html" with table_modal="NetBoxAttachmentTable_config" %}
    {% endblock table_controls %}
            {% htmx_table 'plugins:netbox_attachments:netboxattachment_list' object_type_id=object|content_type_id object_id=object.pk %}
{% endblock content %}
```

The key here is that `table_controls_htmx.html` is included with a `table_modal` variable, which should correspond to `_config`.  This is then intended to be used within the `htmx_table` tag to provide the configuration functionality.

The issue I'm encountering is that this setup works perfectly when the plugin renders as an [**Additional Tab**](https://netboxlabs.com/docs/netbox/en/stable/plugins/development/views/#additional-tabs).  However, when I try to use the same code in a template rendered via [**Extra Template Content**](https://netboxlabs.com/docs/netbox/en/stable/plugins/development/views/#extra-template-content), the "Configure Table" button doesn't function as expected.

It seems the context or the way `htmx_table` interacts with the included template is different depending on whether it's rendered as an Additional Tab or Extra Template Content.

Any insights into why this might be happening and how to resolve it would be greatly appreciated! I suspect the `object` context or the way the template is rendered might be different, but I'm not sure how to address it.
